NumLib/DOF/LocalToGlobalIndexMap.cpp

```diff
@@ -26,16 +27,16 @@ LocalToGlobalIndexMap::findGlobalIndices(
     std::size_t elem_id = 0;
     for (ElementIterator e = first; e != last; ++e, ++elem_id)
     {
-        std::size_t const nnodes = (*e)->getNumberOfNodes();
-
         LineIndex indices;
-        indices.reserve(nnodes);
 
-        for (unsigned n = 0; n < nnodes; n++)
+        for (auto* n = (*e)->getNodes(); n < (*e)->getNodes()+(*e)->getNumberOfNodes(); ++n)
         {
+            // Check if the element's node is in the given list of nodes.
+            if (std::find(std::begin(nodes), std::end(nodes), *n) == std::end(nodes))
```

Comment:
we agreed to keep this function as suggested here and created another function for handling my special case. I'll update my PR after this PR is merged.